app/services/transfer_product.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class TransferProduct < ServiceObject
+  def initialize(*args)
+    super()
+    @product = args[0][:product]
+    @new_owner = args[0][:new_owner]
+    @current_owner = args[0][:current_owner]
+    @error = false
+  end
+
+  def call
+    @status = :error
+    return @error_code = 4207 unless ProductPolicy.new(current_owner, product).transfer_from?
+    return @error_code = 4208 unless ProductPolicy.new(new_owner, product).transfer_to?
+
+    transfer_product
+  end
+
+  private
+
+  attr_reader :new_owner, :product, :current_owner
```

Comment:
We are not accessing to those methods, why do we have those readers? also we have the `error` instance variable that we are using in the controller but we don't have accessors here for it.

Another good practice for accessors is have it at the top of the file.

Reply:
Error variable is inherited from ServiceObject